src/extension.ts

```diff
@@ -152,27 +162,26 @@ async function generateReferenceList(doc: TextDocument): Promise<string[]> {
 function detectReferenceListRange(doc: TextDocument): Range {
   const fullText = doc.getText();
 
-  // find line number of header, and assume 0 for line start
-  // if header is not found, this will be last line of the file
-  const header = [
-    fullText.split(REFERENCE_HEADER)[0].split(docConfig.eol).length - 1,
-    0,
-  ];
-
-  // find line number and char position where footer ends
-  const footer = [
-    fullText.split(REFERENCE_FOOTER)[0].split(docConfig.eol).length,
-    0,
-  ];
-
-  // if header and footer are on the same line, that means we have no references section
-  if (header[0] === footer[0]) {
+  const headerIndex = fullText.indexOf(REFERENCE_HEADER);
+  const footerIndex = fullText.lastIndexOf(REFERENCE_FOOTER);
+
+  if (headerIndex < 0) {
+    return null;
+  }
+
+  const headerLine =
+    fullText.substring(0, headerIndex).split(docConfig.eol).length - 1;
+
+  const footerLine =
+    fullText.substring(0, footerIndex).split(docConfig.eol).length - 1;
+
+  if (headerLine >= footerLine) {
     return null;
   }
 
   return new Range(
-    new Position(header[0], header[1]),
-    new Position(footer[0], footer[1])
+    new Position(headerLine, 0),
+    new Position(footerLine, REFERENCE_FOOTER.length)
```

Comment:
Previously we over-matched to the next line after footer, now we'll only read the footer length